lib/libmbedtls/mbedtls/library/bignum.c

```diff
@@ -127,14 +145,28 @@ int mbedtls_mpi_grow( mbedtls_mpi *X, size_t nblimbs )
 
     if( X->n < nblimbs )
     {
-        if( ( p = (mbedtls_mpi_uint*)mbedtls_calloc( nblimbs, ciL ) ) == NULL )
-            return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
+        if( X->use_mempool )
+	{
+            p = mempool_alloc( mbedtls_mpi_mempool, nblimbs * ciL );
+            if( p == NULL )
+                return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
+            memset( p, 0, nblimbs * ciL );
+        }
+        else
+        {
+            p = (mbedtls_mpi_uint*)mbedtls_calloc( nblimbs, ciL );
```

Comment:
minor: mbedtls tends to put spaces everywhere:
```patch
-            p = (mbedtls_mpi_uint*)mbedtls_calloc( nblimbs, ciL );
+            p = (mbedtls_mpi_uint *) mbedtls_calloc( nblimbs, ciL );
```

Reply:
I'm not so sure about this, here I'm casting in the exact same way as the code which is replaced by this. Normally I wouldn't have casted at all since it's a bit silly to cast a `void *` to another pointer.